### DISCUSSION TITLE
feat: edit an existing event

### DISCUSSION THREAD

**Discussion by scottrigby** [niklasmtj/json-events#17] [Ideas]
so far we have github issue templates for creating new events. But what if we want to edit an existing one? . What's the easiest way for us to allow non-technical end users to do that?

**Comment by scottrigby** [comment]
Unfortunately, pull request templates don't work like this; those are just for PR default intro text.

ideas:

1. What if each row had a primary key, like an auto-generated event ID? (we shouldn't just compare date because there could be more than one event on the same date). We could just let the user optionally specify an existing event ID to override using the same issue method.

1. What about using the issue to create YAML instead of JSON? One file per event? This way, a user could just make a pull request against that YAML file and it's still human readable. This kind of defeats the purpose of using GItHub forms as a CMS, but trade off is you don't need a primary key like event ID.

Any other ideas? Goal is to keep this proof of concept as simple as possible, showing how to use GitHub forms as a simple CMS for end users who don't know anything about JSON or coding necessarily.

**Comment by niklasmtj** [comment]
I think what we can do to edit an issue and so the underlying event is to use the issue number for example #5 as the unique ID of the event. We can get that issue number in the action with `github.event.number`. This way, when we for example update the issue, the action gets triggered again, first checks if the event is already inside of the events structure, if yes replaces the old event with the new information. If it is not found, we will continue as we are doing now. 

We could even do that check on issue creation, this way we can use the same action every time. What do you think? 

Added it in #18
